### PR TITLE
Broadcast team assignment updates via websockets

### PR DESCRIPTION
**Pull request by madwau**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I added multiple integration tests (Spring) related to the features
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I documented the TypeScript code using JSDoc style.

### Motivation and Context
At the moment the participation status is not being updated right away when students are assigned to teams. Even after a tutor has added them into a team, it will still say "You have not been assigned to a team yet" in the exercise card. The new team assignment status will only be reflected after the student refreshes the page. This PR adds broadcasting of team assignment changes via websockets so that students will see right away that they can start working on an exercise as soon as the tutor has created their team.

### Description
- Add `TeamWebsocketService` with method `sendTeamAssignmentUpdate` to broadcast changes in team assignments
- Use this service method when creating, updating, deleting or importing teams in `TeamResource`
- Add observable `teamAssignmentUpdates` to `team.service.ts` that emits team assignment updates
- Subscribe to this observable in the courses view and the course overview
- On receiving a team assignment update: update assigned team id, team participation and participation status on affected exercise

### Steps for Testing
Preconditions: A team-based exercise must exist in a course.

1. Log in as user A to Artemis (needs to be student in course)
2. Navigate to Course Overview
3. Click on a course that has a team-based exercise
4. In a different browser session, log in as user B (needs to be instructor in course)
5. Let user B assign and unassign user A from different teams for the team-based exercise
6. User A should see his participation status changing in the exercise row (from "You have not been assigned to a team yet" to "You have not started this exercise yet", or alternatively "No submission" etc. if a participation already exists, and vice-versa)

### Screencast
Behavior when adding and removing the user from the team:
captured

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -12463,6 +12463,13 @@ tslint-loader@3.6.0:
     rimraf "^2.4.4"
     semver "^5.3.0"
 
+tslint-no-unused-expression-chai@0.1.4:
```
Comment (sleiss) [inline yarn.lock:12466]: Are the changes in this file intended?

Comment (madwau) [reply yarn.lock]: I think the `develop` branch did not have the latest `yarn.lock`, hence the differences that got generated for me locally. I just merged `develop` and now there are no more `yarn.lock` changes.

**Comment by madwau** [comment]
> I've tested the feature and it works as intended, good work!
> 
> Are the changes in the yarn.lock file intended? I could not identify any new usages of the no-unused-expression annotation that would need a change.

Great, thanks for testing! I responsed to the `yarn.lock` changes in the thread.

File: yarn.lock (hunk 1 above)
Comment (sleiss) [reply yarn.lock]: Alright, thanks for looking into it!

**Comment by madwau** [comment]
> Potentially found a bug. Steps to reproduce this:
> 
> 1. Perform the steps as described in the PR. State at the end: User A is assigned to a team.
> 2. User A logs out
> 3. User A logs in
> 4. User A visits the course page with the team exercise
> 5. User B unassigns User A from the team
> 6. User A still sees the state "You have not started this exercise yet" despite not being assigned to a team any longer
> 7. Inspecting your team by clicking on the team button highlighted in the screenshot
>    TeamPage
>    below doesn't load.
> 
> Can you reproduce this?

Thanks, I will try to reproduce it now.

**Pull request merged**